### ISSUE TITLE
FAST002 fix is invalid with `...`

### ISSUE THREAD

**Issue by CoderJoshDK** [astral-sh/ruff#20800]
### Summary

FAST002 converts parameters into their `Annotated` counterparts. If the source code contains an ellipses type (literal `...`) then the fix will generate incorrect semantics.

Take the following signature:

```py
@router.get("/available")
async def get_available(
    ghl_account_id: str = Query(..., description="GHL account identifier"),
) -> SystemsAvailableResponse: ...
```
The use of `...` in the default for `Query` is valid syntax (should be picked up in a lint that you shouldn't do this, but it is "valid"). 
When trying to run fix, you get:

```py
@router.get("/available")
async def get_available(
    ghl_account_id: Annotated[str, Query(description="GHL account identifier")] = ...,
) -> SystemsAvailableResponse: ...
```

When in reality, there is no default value and ruff should convert it to:

```py
@router.get("/available")
async def get_available(
    ghl_account_id: Annotated[str, Query(description="GHL account identifier")],
) -> SystemsAvailableResponse: ...
```

### Version

ruff 0.14.0

**Comment by CoderJoshDK** [comment]
Minimum required repo:

```sh
uv add "fastapi[standard]"
```

```py
# main.py
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/")
async def root(name: str = Query(..., description="Your name!")):
    return {"message": f"Hello {name}"}
```

```sh
uv run fastapi dev main.py
```

If you then visit the docs page, you will find that "name" is marked as required. If you then run the fix:

```py
from typing import Annotated

from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/")
async def root(name: Annotated[str, Query(description="Your name!")] = ...):
    return {"message": f"Hello {name}"}
```

And visit docs now; you find name is no longer required. Implying a default. But executing the endpoint will cause an error
```
ValueError: [TypeError("'ellipsis' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```

So I would consider this fix as incorrect, seeing that it changes the behavior. To match behavior after fix, drop the `...`.

**Comment by ntBre** [comment]
Thanks for the report, this does seem like a bug. At first I thought this was a general type error, but as you said, the ellipsis has special meaning to fastapi, and I think we should handle that in the fix.
